### PR TITLE
Add Facebook Support

### PR DESCRIPTION
**Pull request by HexaCubist**
This is tested against [the linked sample file](https://ericezam.github.io/MessengerDataAnalysis/files/demofile.json) as provided by [Messenger Data Analysis](https://ericezam.github.io/MessengerDataAnalysis/), but not tested with real data yet (still waiting on the download)

Live version available here: https://visioning.zac.nz/

### CODE REVIEW

**Comment by HexaCubist** [comment]
You'll want to get your copy of Facebook Messenger's data by following instructions here: https://www.facebook.com/help/1701730696756992 and then uploading only the json file for the conversation you are looking for.

File: src/math.js [hunk 1]
```diff
@@ -25,6 +25,26 @@ var mm_dd_regex = /^(((0)?[0-9])|((1)[0-2]))(\/)([0-2][0-9]|(3)[0-1])(\/)(\d{2}|
...
+        return msg.type == "Generic" && msg.content
+    }).map( (msg) => {
+        return {
+            'name': msg.sender_name,
+            'date': new Date(msg.timestamp_ms),
+            'ID': `${msg.sender_name}_${msg.timestamp_ms}`,
+            'BODY': msg.content
+        };
+    });
+    // debugger;
```
Comment (BryceStevenWilley) [inline src/math.js]: Does this always just trigger a debugger breakpoint in Chrome?  That's a cool trick, but I don't think it needs to be in there permanently?

File: src/graphs.js [hunk 2]
```diff
@@ -439,7 +439,7 @@ function handleFileSelect(evt) {
         // files is a FileList of File objects. List some properties.
         file_to_read = files[0];
         let f = file_to_read;
-        if (f.type != 'text/plain' && f.type != 'text/csv' &&
+        if (f.type != 'text/plain' && f.type != 'text/csv' && f.type != 'application/json' &&
```
Comment (BryceStevenWilley) [inline src/graphs.js:442]: Could you add "or a Facebook json file" to the alert below?

**Review by BryceStevenWilley — APPROVED** [review]
Looks great, thanks a ton!  The only thing left after this would be documentation, I can handle that though. Is it generally easy to find the specific conversation in the entire FB dump, would someone need instructions to convert from user IDs or whatever?

File: src/math.js (hunk 1 above)
Comment (HexaCubist) [reply src/math.js]: Yeah cool trick, doesn't need to be there though you're right! Was just trying to get a working version as quickly as possible

File: src/graphs.js (hunk 2 above)
Comment (HexaCubist) [reply src/graphs.js:442]: Sure

**Comment by BryceStevenWilley** [comment]
Merged! Thanks @HexaCubist!

**Pull request merged**